presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java

```diff
@@ -66,7 +68,12 @@
     private long hashCollisions;
     private double expectedHashCollisions;
 
-    public BigintGroupByHash(int hashChannel, boolean outputRawHash, int expectedSize)
+    // reserve enough memory before rehash
+    private final UpdateMemory updateMemory;
+    private long reservedMemoryInBytes;
```

Comment:
This is not a good name. When I read "Reserved", my immediate understanding is that this include all memory usage by this object．